src/osd/PG.cc

```diff
@@ -7818,14 +7896,19 @@ PG::PriorSet::PriorSet(bool ec_pool,
   for (unsigned i = 0; i < acting.size(); i++) {
     if (acting[i] != CRUSH_ITEM_NONE)
       probe.insert(pg_shard_t(acting[i], ec_pool ? shard_id_t(i) : shard_id_t::NO_SHARD));
+      request_log.insert(pg_shard_t(acting[i], ec_pool ?
+        shard_id_t(i) : shard_id_t::NO_SHARD));
   }
   // It may be possible to exclude the up nodes, but let's keep them in
   // there for now.
   for (unsigned i = 0; i < up.size(); i++) {
     if (up[i] != CRUSH_ITEM_NONE)
       probe.insert(pg_shard_t(up[i], ec_pool ? shard_id_t(i) : shard_id_t::NO_SHARD));
+      request_log.insert(pg_shard_t(up[i], ec_pool ?
+        shard_id_t(i) : shard_id_t::NO_SHARD));
   }
```

Comment:
This is concerning.  Currently, in the common case where one osd went down, the other osds in the acting set won't need to send any actual log entries.  With this change, every replica from every osd which shares a pg with the down osd will have to send the *entire* log to the primary.  That seems unlikely to be a good tradeoff.  Is there a way to guess at how much log will be required and only request that much?